### ISSUE TITLE
debug: true config option not printing queries run through QueryBuilder.getSingleResult()

### ISSUE THREAD

**Issue by gerardolm** [mikro-orm/mikro-orm#3681]
**Describe the bug**
Using version 5.5.0, `debug: true` in config prints queries generated through generic repository methods, but not through `QueryBuilder.getSingleResult()`.

**To Reproduce**
1. Use `QueryBuilder.getSingleResult()` in a test with `debug: true` in MikroORM options object.
2. Expect query to be logged.
3. Wait forever.

**Expected behavior**
Queries are logged.

**Additional context**
Redacted code not triggering the print:

```typescript
await this.em
      .createQueryBuilder(SomeEntity, 's')
      .select([
        's.*',
        someSubQuery.as('an_alias'),
      ])
      .where(where)
      .getSingleResult()
```

This is run within a custom repository. Queries run 100% because I am able to read the result.

**Versions**

| Dependency | Version |
| - | - |
| node | 14.19.0 (I know...) |
| typescript | 4.7.4 |
| mikro-orm | 5.5.0 |
| mikro-orm/postgresql | 5.5.0 |

**Comment by B4nan** [comment]
Not reproducible, and tbh I doubt it would behave differently. Logging is implemented on the connection level, and QB uses the exact same API. In fact, some of the queries from UoW are done by executing QB instance directly.

Let's reopen with a complete repro so it's clear what you are doing.

> node | 14.19.0 (I know...)

Hard to believe, as MikroORM v5.5 is most probably not compatible unless you install some shims for missing language features like `??=` operator (not sure if that's even possible).

**Comment by gerardolm** [comment]
Rightfully closed.

Other queries were also not being logged (simple inserts through generic repo methods) even though they were run earlier in the same test case, and I hadn't noticed that (I was comparing against another integration test with several cases where queries were logged).

The case I reported started logging the queries when I added another integration test case to the same file, and they stop logging again when there is only one. This could just be a symptom of the logging being deferred and the test run ending before the logging actually happens. Apologies for being trigger-happy with the issue.

And for the record: yes, I am definitely running that version of node. Happy to run any check you want.
